### PR TITLE
Allow messaging "non-contactable" patients if needed

### PR DESCRIPTION
**Pull request by harimohanraj89**
**Story card:** https://app.clubhouse.io/simpledotorg/story/297/bangladesh-sms-blast

## Because

BD Phone numbers are not all marked as "mobile" numbers.

## This addresses

Adding a `contactable` flag to the `MessagePatients` service that does two things:

* Decides whether to scope the passed-in `patients` with the `contactable` scope, and
* Decides whether to use the patient's latest _mobile_ number, or phone number.

### CODE REVIEW

File: lib/tasks/scripts/message_patients.rb [hunk 1]
```diff
@@ -59,7 +60,19 @@ def send_messages
...
+    else
+      patients
+    end
+  end
+
+  def phone_number_for(patient)
+    if contactable
+      patient.latest_mobile_number
+    else
+      patient.latest_phone_number
```
Comment (prabhanshuguptagit) [inline lib/tasks/scripts/message_patients.rb:74]: What happens if the patient doesn't have a phone number? Is that a possible case?

File: lib/tasks/scripts/message_patients.rb [hunk 2]
```diff
@@ -3,13 +3,14 @@ def self.call(*args)
     new(*args).call
   end
 
-  attr_reader :patients, :message, :channel, :dryrun, :verbose, :report
+  attr_reader :patients, :message, :channel, :contactable, :dryrun, :verbose, :report
   VALID_CHANNELS = [:whatsapp, :sms]
 
-  def initialize(patients, message, channel: :whatsapp, dryrun: false, verbose: true)
+  def initialize(patients, message, channel: :whatsapp, contactable: true, dryrun: false, verbose: true)
```
Comment (prabhanshuguptagit) [inline lib/tasks/scripts/message_patients.rb:9]: `only_contactable: true`?

File: lib/tasks/scripts/message_patients.rb (hunk 1 above)
Comment (harimohanraj89) [reply lib/tasks/scripts/message_patients.rb:74]: Great call. In that case we should most likely no-op the patient. I'll put that in 👍

**Comment by harimohanraj89** [comment]
Thanks for the review. Comments addressed in 468f8aa

**Pull request merged**